pandas/plotting/_matplotlib/core.py

```diff
@@ -1196,10 +1196,10 @@ def _post_plot_logic(self, ax, data):
         from matplotlib.ticker import FixedLocator
 
         def get_label(i):
-            try:
-                return pprint_thing(data.index[i])
-            except Exception:
+            if i >= len(data.index):
+                # TODO: is getting here indicative of a larger problem?
```

Comment:
@datapythonista I think you've worked in this area recently, any idea if this is something to worry about?

Reply:
Is there a test (or example code) that ended up here?

Reply:
Didn't work in this exact part, but this seems reasonable. I'm assuming this only affects the values in the axis, and it removes the decimals if they are zero, right?

I'm wondering if instead of `i == int(i)` it'd be better to use `i.apply(float.is_integer)`.

Reply:
i had completely forgotten about float.is_integer; neat.  But I don't think we in general have an `apply` do we?